src/components/NcAppSidebar/NcAppSidebarTabs.vue

```diff
@@ -216,64 +210,38 @@ export default {
 		 */
 		updateActive() {
 			this.activeTab = this.active
-				&& this.tabs.findIndex(tab => tab.id === this.active) !== -1
+			&& this.tabs.findIndex(tab => tab.id === this.active) !== -1
 				? this.active
 				: this.tabs.length > 0
 					? this.tabs[0].id
 					: ''
 		},
 
 		hasMdIcon(tab) {
-			return tab?.$slots?.icon
+			return tab?.$slots?.icon || tab?.$scopedSlots?.icon
 		},
 
 		/**
-		 * Manually update the sidebar tabs according to $slots.default
+		 * Register child tab in the tabs
+		 *
+		 * @param {object} tab - tab props (only the "id" is used actually)
 		 */
-		updateTabs() {
-			if (!this.$slots.default) {
-				this.tabs = []
-				return
-			}
-
-			// Find all valid children (AppSidebarTab, other components, text nodes, etc.)
-			const children = this.$slots.default.filter(elem => elem.tag || elem.text.trim())
-
-			// Find all valid instances of AppSidebarTab
-			const invalidTabs = []
-			const tabs = children.reduce((tabs, tabNode) => {
-				const tab = tabNode.componentInstance
-				// Make sure all required props are provided and valid
-				if (IsValidString(tab?.name)
-					&& IsValidStringWithoutSpaces(tab?.id)
-					&& (IsValidStringWithoutSpaces(tab?.icon) || tab?.$slots?.icon)) {
-					tabs.push(tab)
-				} else {
-					invalidTabs.push(tabNode)
-				}
-				return tabs
-			}, [])
-
-			// Tabs are optional, but you can use either tabs or non-tab-content only
-			if (tabs.length !== 0 && tabs.length !== children.length) {
-				Vue.util.warn('Mixing tabs and non-tab-content is not possible.')
-				invalidTabs.map(invalid => console.debug('Ignoring invalid tab', invalid))
-			}
-
-			// We sort the tabs by their order or by their name
-			this.tabs = tabs.sort((a, b) => {
-				const orderA = a.order || 0
-				const orderB = b.order || 0
-				if (orderA === orderB) {
-					return OC.Util.naturalSortCompare(a.name, b.name)
-				}
-				return orderA - orderB
-			})
+		registerTab(tab) {
+			this.tabs.push(tab)
```

Comment:
We could still validate the tabs before pushing them to the `tabs` array like above:
```js
				if (IsValidString(tab?.name)
					&& IsValidStringWithoutSpaces(tab?.id)
					&& (IsValidStringWithoutSpaces(tab?.icon) || tab?.$slots?.icon))
```

Reply:
We could do the sorting here already. Then we don't need the computed prop.